libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5170,10 +5158,37 @@ void GCS_MAVLINK::convert_COMMAND_LONG_to_COMMAND_INT(const mavlink_command_long
     out.param2 = in.param2;
     out.param3 = in.param3;
     out.param4 = in.param4;
-    const bool stores_location = command_long_stores_location((MAV_CMD)in.command);
-    out.x = convert_COMMAND_LONG_loc_param(in.param5, stores_location);
-    out.y = convert_COMMAND_LONG_loc_param(in.param6, stores_location);
+
+    // Spec calls for NaN in COMMAND_LONG for invalid, INT32_T-max in
+    // COMMAND_INT:
+    float param5 = in.param5;
+    if (isnan(param5)) {
+        param5 = INT32_MAX;
+    }
+    float param6 = in.param6;
+    if (isnan(param6)) {
+        param6 = INT32_MAX;
+    }
+
+    if (!isfinite(param5) || !isfinite(param6)) {
+        return MAV_RESULT_DENIED;
+    }
+
+    if (command_long_stores_location((MAV_CMD)in.command)) {
+        // check parameters are valid latitude/longitude:
+        if (!check_lat(param5) || !check_lng(param6)) {
+            return MAV_RESULT_DENIED;
+        }
+
+        out.x = param5 * 1e7;
+        out.y = param6 * 1e7;
+    } else {
+        out.x = param5;
```

Comment:
we need to see if there may be any commands that relied on the previous zero, and now will get INT32_MAX, motor test is a possible issue